awscli/customizations/ecs/deploy.py

```diff
@@ -282,9 +291,30 @@ def update_task_def_arn(self, new_arn):
         appspec_obj[resources_key] = updated_resources
         self._appspec_dict = appspec_obj
 
-    def wait_for_deploy_success(self, id):
+    def wait_for_deploy_success(self, id, wait_min):
         waiter = self._client.get_waiter("deployment_successful")
-        waiter.wait(deploymentId=id)
+        wait_msg = ""
+
+        if wait_min <= 30 or wait_min > MAX_WAIT_MIN:
```

Comment:
If I'm understanding this correctly, if the `wait_min` is too low or too high we use the default waiter settings?
I would think if the max wait is too long, we would cap at the max wait. If it's too low we would use the min. 

I would think it works like this:
If there was no custom wait time information from the config we would use the default waiter settings.
If there is custom wait time information we would validate it based on the min/max and then use a customer waiter configuration. 

It's worth noting that if the default waiter configuration were to ever change the min of 30 here might become out of sync.

Reply:
Capping at max wait time makes sense. Also good point re: if the default waiter changes. I can change it so we're always calling `waiter.Wait()` with a `WaiterConfig`